src/OpenTelemetry.ResourceDetectors.ProcessRuntime/ProcessRuntimeDetector.cs

```diff
@@ -0,0 +1,54 @@
+// <copyright file="ProcessRuntimeDetector.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+#if !NETFRAMEWORK
+using System;
+#endif
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+using OpenTelemetry.Resources;
+
+namespace OpenTelemetry.ResourceDetectors.ProcessRuntime;
+
+/// <summary>
+/// Process runtime detector.
+/// </summary>
+public class ProcessRuntimeDetector : IResourceDetector
+{
+    /// <summary>
+    /// Detects the resource attributes from .NET runtime.
+    /// </summary>
+    /// <returns>Resource with key-value pairs of resource attributes.</returns>
+    public Resource Detect()
+    {
+#if NETFRAMEWORK
+        var frameworkDescriptionParts = RuntimeInformation.FrameworkDescription.Split(' ');
+        var netFrameworkVersion = frameworkDescriptionParts.Length > 1 ? frameworkDescriptionParts[frameworkDescriptionParts.Length - 1] : "unknown";
+#endif
+
+        return new Resource(new List<KeyValuePair<string, object>>(3)
+        {
+            new(ProcessRuntimeSemanticConventions.AttributeProcessRuntimeDescription, RuntimeInformation.FrameworkDescription),
+#if NETFRAMEWORK
+            new(ProcessRuntimeSemanticConventions.AttributeProcessRuntimeName, ".NET Framework"),
+            new(ProcessRuntimeSemanticConventions.AttributeProcessRuntimeVersion, netFrameworkVersion),
+#else
+            new(ProcessRuntimeSemanticConventions.AttributeProcessRuntimeName, ".NET"),
```

Comment:
Could we use the `RuntimeInformation.FrameworkDescription` again here? https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.frameworkdescription?view=net-8.0#remarks

I think it would be good to have `.NET Core` and `.NET Native` as the tag values.